balancer-js/src/modules/data/token-prices/coingecko.ts

```diff
@@ -98,10 +101,61 @@ export class CoingeckoPriceRepository implements Findable<Price> {
     return promise;
   }
 
+  private fetchHistorical(
```

Comment:
i think it would be best to separate historical use-case into separate repository. both can use the same coingecko client.

Reply:
please, check my latest change

I split the provider in 2 services

Reply:
great, thank you!